test/integration/app.js

```diff
@@ -151,42 +144,85 @@ describe('The results page', () => {
   });
 
   describe('error handling', () => {
-    const notFoundResponse = getSampleResponse('postcodesio-responses/404.json');
-    const invalidPostcodePassingRegex = 'LS0';
+    describe('general site', () => {
+      it('should handle page not found.', (done) => {
+        chai.request(server)
+          .get('/finders/unknown')
+          .end((err, res) => {
+            expect(err).to.not.be.null;
+            expect(res).to.have.status(404);
+            expect(res).to.be.html;
+            const $ = cheerio.load(res.text);
+            expect($('.local-header--title--question').text().trim())
+              .to.equal('Page not found');
+            done();
+          });
+      });
+    });
+    describe('postcode lookup', () => {
+      describe('invalid postcodes', () => {
+        const notFoundResponse = getSampleResponse('postcodesio-responses/404.json');
+        const invalidPostcodePassingRegex = 'LS0';
+
+        it('should lookup a syntactically valid but unknown postcode and return an error message',
+          (done) => {
+            const postcodesioScope =
+              nock('https://api.postcodes.io')
+              .get(`/outcodes/${invalidPostcodePassingRegex}`)
+              .times(1)
+              .reply(404, notFoundResponse);
+            chai.request(server)
+              .get(resultsRoute)
+              .query({ location: invalidPostcodePassingRegex })
+              .end((err, res) => {
+                checkHtmlResponse(err, res);
+                expect(res.text).to
+                  .contain(messages.invalidPostcodeMessage(invalidPostcodePassingRegex));
+                expect(postcodesioScope.isDone()).to.equal(true);
+                done();
+              });
+          });
 
-    const postcodesioScope =
-    nock('https://api.postcodes.io')
-      .get(`/outcodes/${invalidPostcodePassingRegex}`)
-      .times(1)
-      .reply(404, notFoundResponse);
+        it('should validate the postcode and return an error message', (done) => {
+          nock('https://api.postcodes.io')
```

Comment:
There should be no requests going to postcodes.io as the validation will be done by the postcode regex checker. The `nock` can be removed.